File: src/main/java/org/chousik/collection/Person.java
package org.chousik.collection;

import lombok.Getter;
import lombok.Setter;

import java.util.Objects;

/**
 * Класс персона
 */
@Getter
@Setter
public class Person implements Comparable<Person> {
    private String name; //Поле не может быть null, Строка не может быть пустой
    private String passportID; //Строка не может быть пустой, Длина строки должна быть не меньше 6, Поле не может быть null
    private Color hairColor; //Поле не может быть null
    private Country nationality; //Поле может быть null
    private Location location; //Поле не может быть null

    public Person() {
    }

    public Person(String name, String PassportID, Color color, Country country, Location location) {
        this.name = name;
        this.passportID = PassportID;
        this.location = location;
        this.hairColor = color;
        this.nationality = country;
    }

    @Override
    public int compareTo(Person o) {
        return this.nationality.ordinal() - o.nationality.ordinal();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Person person = (Person) o;
        return Objects.equals(name, person.name) && Objects.equals(passportID, person.passportID) && hairColor == person.hairColor && nationality == person.nationality && Objects.equals(location, person.location);
    }

    @Override
    public int hashCode() {
        return Objects.hash(name, passportID, hairColor, nationality, location);
    }

    @Override
    public String toString() {
        return "Person{" +
                "name='" + name + '\'' +
                ", passportID='" + passportID + '\'' +
                ", hairColor=" + hairColor.toString() +
                ", nationality=" + nationality.toString() +
                ", location=" + location.toString() +
                '}';
    }
}

File: src/main/kotlin/collection/Person.kt
package org.chousik.collection

import java.util.*
class Person(
    private var name: String,
    private var passportID: String,
    private var hairColor: Color,
    private var nationality: Country,
    private var location: Location
) : Comparable<Person> {

    override fun compareTo(other: Person): Int {
        return nationality.ordinal - other.nationality.ordinal
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val person = other as Person
        return name == person.name && passportID == person.passportID && hairColor == person.hairColor && nationality == person.nationality && location == person.location
    }

    override fun hashCode(): Int {
        return Objects.hash(name, passportID, hairColor, nationality, location)
    }

    override fun toString(): String {
        return "Person{" +
                "name='" + name + '\'' +
                ", passportID='" + passportID + '\'' +
                ", hairColor=" + hairColor.toString() +
                ", nationality=" + nationality.toString() +
                ", location=" + location.toString() +
                '}'
    }
}